### PR TITLE
Api interfaces mark 1

### LINKED ISSUE #32024: Mark Framework interfaces as API part 1
The following interfaces should be marked as API in 2.4-develop branch

1. lib/internal/Magento/Framework/Amqp/Topology/BindingInstallerInterface.php
1. lib/internal/Magento/Framework/Api/AttributeTypeResolverInterface.php
1. lib/internal/Magento/Framework/Api/CriteriaInterface.php
1. lib/internal/Magento/Framework/Api/Search/AggregationInterface.php
1. lib/internal/Magento/Framework/Api/Search/AggregationValueInterface.php
1. lib/internal/Magento/Framework/Api/Search/BucketInterface.php
1. lib/internal/Magento/Framework/Api/Search/DocumentInterface.php
1. lib/internal/Magento/Framework/Api/Search/ReportingInterface.php
1. lib/internal/Magento/Framework/Api/SearchCriteria/CollectionProcessor/ConditionProcessor/CustomConditionInterface.php
1. lib/internal/Magento/Framework/Api/SearchCriteria/CollectionProcessor/ConditionProcessor/CustomConditionProviderInterface.php
1. lib/internal/Magento/Framework/Api/SimpleBuilderInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpConnectActionInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpDeleteActionInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpGetActionInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpOptionsActionInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpPatchActionInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpPostActionInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpPropfindActionInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpPutActionInterface.php
1. lib/internal/Magento/Framework/App/Action/HttpTraceActionInterface.php
1. lib/internal/Magento/Framework/App/AreaInterface.php
1. lib/internal/Magento/Framework/App/Config/CommentInterface.php
1. lib/internal/Magento/Framework/App/Config/CommentParserInterface.php
1. lib/internal/Magento/Framework/App/Config/ConfigResource/ConfigInterface.php
1. lib/internal/Magento/Framework/App/Config/ConfigSourceInterface.php
1. lib/internal/Magento/Framework/App/Config/ConfigTypeInterface.php
1. lib/internal/Magento/Framework/App/Config/Reader/Source/SourceInterface.php
1. lib/internal/Magento/Framework/App/Config/Scope/ReaderInterface.php
1. lib/internal/Magento/Framework/App/Config/Spi/PostProcessorInterface.php
1. lib/internal/Magento/Framework/App/Config/Spi/PreProcessorInterface.php

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Marked following interfaces as API

lib/internal/Magento/Framework/Amqp/Topology/BindingInstallerInterface.php
lib/internal/Magento/Framework/Api/AttributeTypeResolverInterface.php
lib/internal/Magento/Framework/Api/CriteriaInterface.php
lib/internal/Magento/Framework/Api/Search/AggregationInterface.php
lib/internal/Magento/Framework/Api/Search/AggregationValueInterface.php
lib/internal/Magento/Framework/Api/Search/BucketInterface.php
lib/internal/Magento/Framework/Api/Search/DocumentInterface.php
lib/internal/Magento/Framework/Api/Search/ReportingInterface.php
lib/internal/Magento/Framework/Api/SearchCriteria/CollectionProcessor/ConditionProcessor/CustomConditionInterface.php
lib/internal/Magento/Framework/Api/SearchCriteria/CollectionProcessor/ConditionProcessor/CustomConditionProviderInterface.php
lib/internal/Magento/Framework/Api/SimpleBuilderInterface.php
lib/internal/Magento/Framework/App/Action/HttpConnectActionInterface.php
lib/internal/Magento/Framework/App/Action/HttpDeleteActionInterface.php
lib/internal/Magento/Framework/App/Action/HttpGetActionInterface.php
lib/internal/Magento/Framework/App/Action/HttpOptionsActionInterface.php
lib/internal/Magento/Framework/App/Action/HttpPatchActionInterface.php
lib/internal/Magento/Framework/App/Action/HttpPostActionInterface.php
lib/internal/Magento/Framework/App/Action/HttpPropfindActionInterface.php
lib/internal/Magento/Framework/App/Action/HttpPutActionInterface.php
lib/internal/Magento/Framework/App/Action/HttpTraceActionInterface.php
lib/internal/Magento/Framework/App/AreaInterface.php
lib/internal/Magento/Framework/App/Config/CommentInterface.php
lib/internal/Magento/Framework/App/Config/CommentParserInterface.php
lib/internal/Magento/Framework/App/Config/ConfigResource/ConfigInterface.php
lib/internal/Magento/Framework/App/Config/ConfigSourceInterface.php
lib/internal/Magento/Framework/App/Config/ConfigTypeInterface.php
lib/internal/Magento/Framework/App/Config/Reader/Source/SourceInterface.php
lib/internal/Magento/Framework/App/Config/Scope/ReaderInterface.php
lib/internal/Magento/Framework/App/Config/Spi/PostProcessorInterface.php
lib/internal/Magento/Framework/App/Config/Spi/PreProcessorInterface.php

References class/interfaces also marked api
lib/internal/Magento/Framework/MessageQueue/Topology/Config/ExchangeConfigItem/BindingInterface.php

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32024

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
rebased 2.5-develop

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Fixed static tests
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests, Functional Tests B2B

File: lib/internal/Magento/Framework/MessageQueue/Topology/Config/ExchangeConfigItem/BindingInterface.php [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 /**
  * Instances of this interface represent config binging items declared in etc/queue_topology.xsd
+ *
+ * @api
```
Comment (hws47a) [inline lib/internal/Magento/Framework/MessageQueue/Topology/Config/ExchangeConfigItem/BindingInterface.php:11]: Hi @mrtuvn,

Could you please share why this class is also marked as `@api`?

Comment (mrtuvn) [reply lib/internal/Magento/Framework/MessageQueue/Topology/Config/ExchangeConfigItem/BindingInterface.php:11]: https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Amqp/Topology/BindingInstallerInterface.php <= this class need marked api as purpose of this PR. When you see content this interface you can find this interface reference to another Magento\Framework\MessageQueue\Topology\Config\ExchangeConfigItem\BindingInterface (non-api)
So we also need mark this interface referenced to api too

Comment (mrtuvn) [reply lib/internal/Magento/Framework/MessageQueue/Topology/Config/ExchangeConfigItem/BindingInterface.php:11]: You can see info interface listed in description

**Review by hws47a — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @hws47a, thank you for the review.
[ENGCOM-8802](https://jira.corp.magento.com/browse/ENGCOM-8802) has been created to process this Pull Request
:eight_spoked_asterisk: @hws47a, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by mrtuvn** [comment]
Re-targeted PR branch to 2.4-develop as internal discussed with @gabrieldagama

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests

**Comment by gabrieldagama** [comment]
@magento import code to https://github.com/magento-engcom/magento2ce

**Comment by magento-engcom-team** [comment]
@gabrieldagama the branch with code successfully imported into`magento-engcom/magento2ce` repository. Branch name: `imported-magento-magento2-32135`.

**Pull request merged**